src/app/ngrx/effects/castle.effects.ts

```diff
@@ -0,0 +1,64 @@
+import { Injectable } from '@angular/core';
+import 'rxjs/add/operator/switchMap';
+import { Observable } from 'rxjs/Observable';
+import { Actions, Effect } from '@ngrx/effects';
+import { Store } from '@ngrx/store';
+
+import { ActionTypes,
+  CastlePodcastAllTimeMetricsLoadAction, CastlePodcastAllTimeMetricsLoadPayload,
+  CastlePodcastAllTimeMetricsSuccessAction, CastlePodcastAllTimeMetricsFailureAction,
+  CastleEpisodeAllTimeMetricsLoadAction, CastleEpisodeAllTimeMetricsLoadPayload,
+  CastleEpisodeAllTimeMetricsSuccessAction, CastleEpisodeAllTimeMetricsFailureAction } from '../actions';
+import { selectPodcasts } from '../reducers';
+import { filterPodcasts } from '../../shared/util/metrics.util';
+import { CastleService } from '../../core';
+import { PodcastModel } from '../';
+
+@Injectable()
+export class CastleEffects {
+  podcasts: PodcastModel[] = [];
+
+  @Effect()
+  loadAllTimePodcastMetrics = this.actions$
+    .ofType(ActionTypes.CASTLE_PODCAST_ALL_TIME_METRICS_LOAD)
+    .map((action: CastlePodcastAllTimeMetricsLoadAction) => action.payload)
+    .switchMap((payload: CastlePodcastAllTimeMetricsLoadPayload) => {
+      const podcast = filterPodcasts(payload.filter, this.podcasts);
+      if (podcast) {
```

Comment:
(see other comment -- sometimes this.podcasts is not yet populated, so this sends a "No podcasts yet" error and doesn't retry. Wasn't sure whether the best way to handle would be add retry logic or just dispatch the Load action again from the component -- went with the latter)